Breeding-Insight/bi-api
# [BI-2203] Insert sandbox email into bi_user table

# Description
[BI-2203](https://breedinginsight.atlassian.net/browse/BI-2203) Insert sandbox email into bi_user table

_Please include a summary of the change that was made._

- Created 2 migrations
1. A java migration to populate the email of the "BI-DEV Admin" user (based on whether it should be authorized with sandbox Orcid or the production Orcid).
2. A migration to add a constraint on the _bi_user_ table (that prevents the orcid id from being non-null if the email is null).
- Added a configuration variable (ORCID_SANDBOX_AUTHENTICATION) that is true if the Oricid Sandbox should be used, and _false_ infthe production Orcid should be used.

# Testing

- verify that the migration inserts the address in the table
-  sign in to Deltabreed with sys admin permissions and add a new user to the system with [bidevteam@mailinator.com](mailto:bidevteam@mailinator.com) as the address and confirm that this is prevented and an error message is returned saying that a user with that address already exists.


# Checklist:
